nested/network-vnet-ddos.json

```diff
@@ -0,0 +1,56 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "moodleCommon": {
+            "metadata": {
+                "description": "Common Moodle values"
+            },
+            "type": "object"
+        },
+        "vnetName": {
+            "metadata": {
+                "description": "The name of the vnet to which the DDoS protection plan shoud be associated"
+            },
+            "type": "string"
+        },
+        "vNetAddressSpace": {
+            "metadata": {
+                "description": "The vNet Address Space to which the DDoS protection plan shoud be associated"
+            },
+            "type": "string"
+        }
+    },
+    "resources": [
+        {
+            "apiVersion": "2018-02-01",
+            "condition": "[parameters('moodleCommon').ddosSwitch]",
+            "location": "[resourceGroup().location]",
+            "name": "[parameters('moodleCommon').ddosPlanName]",
+            "type": "Microsoft.Network/ddosProtectionPlans"
+        },
+        {
+            "apiVersion": "2018-02-01",
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/ddosProtectionPlans', parameters('moodleCommon').ddosPlanName)]"
+            ],
+            "location": "[resourceGroup().location]",
+            "name": "[parameters('vnetName')]",
```

Comment:
This is great. So you did confirm that this works regardless of whether this vnet already exists or is to be created, right?

Reply:
Yes, I've ran 4 tests 
1. existing vNet and ddosSwitch equal to true
2. existing vNet and ddosSwitch equal to false
3. no existing vNet and ddosSwitch equal to true
4. no existing vNet and ddosSwitch equal to false
Each one of these 4 deployment succeed and I can browse the homepage.

Reply:
💯

Reply:
I have added documentation in the Parameters.md file and the Manage.md file.

Reply:
Thank you so much for your quick follow-up. This is super! I'm merging this PR right after. Thanks again!

Reply:
You're welcome. Thanks for the great work you're doing !

Reply:
Is DDoS protection included with all deployments automatically?